### PR TITLE
Varied bugfixes

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: cmd/incusd/networks.go [hunk 1]
```diff
@@ -489,6 +489,19 @@ func networksPost(d *Daemon, r *http.Request) response.Response {
...
 		}
 
+		n, err := network.LoadByName(s, projectName, req.Name)
+		if err != nil {
+			return response.SmartError(fmt.Errorf("Failed loading network: %w", err))
+		}
+
+		err = s.Authorizer.AddNetwork(r.Context(), projectName, req.Name)
+		if err != nil {
+			logger.Error("Failed to add network to authorizer", logger.Ctx{"name": req.Name, "project": projectName, "error": err})
```
Comment (hallyn) [inline cmd/incusd/networks.go:499]: So you don't want to return an error in this case?

File: client/incus_oidc.go [hunk 2]
```diff
@@ -147,6 +148,40 @@ func (o *oidcClient) do(req *http.Request) (*http.Response, error) {
 	return resp, nil
 }
 
+// dial function executes a websocket request and handles OIDC authentication and refresh.
+func (o *oidcClient) dial(dialer websocket.Dialer, uri string, req *http.Request) (*websocket.Conn, *http.Response, error) {
+	conn, resp, err := dialer.Dial(uri, req.Header)
+	if err != nil && resp == nil {
```
Comment (hallyn) [inline client/incus_oidc.go:154]: So what if err != nil but resp != nil ?  You will ignore the error in that case.  Is that the right thing?

Comment (stgraber) [reply client/incus_oidc.go:154]: Yeah because that's specifically the case we want to retry.

Comment (stgraber) [reply client/incus_oidc.go:154]: The special logic in this function is to handle cases where we get err != nil (Bad handshake) but have a resp filled in with the response code (403 - Unauthorized) and headers (OIDC issuer and client id).

When that happens, we refresh the OIDC token and then re-issue the dialer.Dial with the new token.

**Review by hallyn — APPROVED** [review]

**Pull request merged**